### ISSUE TITLE
Problem with SMTP connection

### ISSUE THREAD

**Issue by chilly2go** [pentacent/keila#227]
Hi,
I'm running my own mailserver with mailcow for multiple years now.

Trying to send mailings with keila through my server proved to be non-functional.
I tried all combinations for ports / tls.

what I got was:
```
postfix/submission/smtpd[99996]: connect from unknown
postfix/submission/smtpd[99996]: SSL_accept error from unknown
postfix/submission/smtpd[99996]: warning: TLS library problem: error:1408F10B:SSL routines:ssl3_get_record:wrong version number:../ssl/record/ssl3_record.c:331:
postfix/submission/smtpd[99996]: lost connection after STARTTLS from unknown
```
now after enabling TLS 1.0 and 1.1 again I got a connect and nothing more. on the server side.
after 10 minutes keila logs:

`[warning] Failed sending email to target@testmail for campaign nmc_GQL5WVpk: {:retries_exceeded, {:network_failure, 'my.server.ip', {:error, :closed}}}`

This looks to me as having two smtp related problems:
- Outgoing connections use old / insecure encryption protocol version
- SMTP sending seems to have general problems (other connection attempts trigger more logging on the mail server that atleast results in a PREGREET message)

I do hope you can reproduce this.

Docker just tells me that I'm running the image tagged "latest" created "2 weeks ago"

**Comment by wmnnd** [comment]
Hey there, thanks for reporting this. Have you been able to resolve this problem in the meantime?

**Comment by chilly2go** [comment]
Nope. 
In the mean time I tried sendgrid but the log only shows:
{"errors" => [%{"field" => nil, "help" => nil, "message" => "The provided authorization grant is invalid, expired, or revoked"}]}

Showing campaign statistics is no help at all as the amount of information shown is exactly 0
No warning or error notification or recipients with delivery problems. Nothing

**Comment by chilly2go** [comment]
I also tried sendgrids smtp relay... 

Port 25 without TLS
 {:retries_exceeded, {:network_failure, '52.57.139.126', {:error, :closed}}}

Port 587 with TLS
 {:retries_exceeded, {:network_failure, '18.197.194.208', {:error, {:options, :incompatible, [verify: :verify_peer, cacerts: :undefined]}}}}

**Comment by chilly2go** [comment]
With my own mailserver. Yes
With sendgrid. No

**Comment by chilly2go** [comment]
Port 587 without TLS
{:retries_exceeded, {:network_failure, '18.197.194.208', {:error, :closed}}}

**Comment by enzokabella** [comment]
We have the same problem. 
12:36:11.388 [warning] Failed sending email to XXX@XXX.de for campaign nmc_neLWGkLw: {:retries_exceeded, {:network_failure, '80.237.132.207', {:error, {:options, :incompatible, [verify: :verify_peer, cacerts: :undefined]}}}}

no emails are going out.
We use the keila docker container.

**Comment by chilly2go** [comment]
Currently keila is unusable for me. sadly
I'm moving to a different solution

**Comment by wmnnd** [comment]
Thank you all for reporting this issue. 

I think this issue is caused by recent changes in Erlang/OTP, but I’m still investigating.

**Comment by wmnnd** [comment]
This issue should be fixed in the 0.12.6 release.
